### ISSUE TITLE
Configure node resolver

### ISSUE THREAD

**Issue by binarymist** [benmosher/eslint-plugin-import#1131]
I'm not exactly sure how this is supposed to work, but it appears as though I'm doing things the right way.

In my .eslinrc.js I have:

```
module.exports = {
  ...
  env: {
    'node': true
  },
  settings: {
    'import/resolver': {
      'node': {
        'paths': ['./config'],
        'moduleDirectories': ['node_modules']
      }
    }    
  }
};
```
but eslint is still complaining: 

`error  Unable to resolve path to module 'config/config'                                         import/no-unresolved`

on a file that has `const config = require('config/config');` at the top.

I'm using `app-module-path` for my package deps.

Opened issue from [#506](https://github.com/benmosher/eslint-plugin-import/issues/506#issuecomment-402386173)

Same deal for `./src` in the rood dir. `package.json` is in the root dir. Have tried adding all the incantations to the `paths` and `moduleDirectories` arrays.

**Comment by binarymist** [comment]
Adding the `'import/no-unresolved': [{commonjs: true}]` to the `.eslingrc.js` removes the error, I'm not exactly sure what's going on here though. I'm not sure what the `2` is for in the `import/no-unresolved: [2, {commonjs: true, amd: true}]`, it doesn't appear to be documented.

**Comment by ljharb** [comment]
It means “error”, and it’s how all eslint rules are configured (with a severity of 0/1/2 or off/warn/error)

By adding it without the 2, you’re disabling the rule, which is unlikely to be what you want.

Can you share your entire eslintrc, with nothing elides?

**Comment by binarymist** [comment]
Ah, OK, thanks.
[.eslingrc.js source](https://github.com/binarymist/purpleteam-orchestrator/blob/frontEndIntegration/.eslintrc.js)

**Comment by ljharb** [comment]
The docs say that “paths” must be an absolute path - not a relative one. https://github.com/benmosher/eslint-plugin-import/tree/master/resolvers/node

**Comment by binarymist** [comment]
Ah, tried that.

**Comment by ljharb** [comment]
In that case, I'm not really sure (but i don't know what `app-module-path` is).

**Comment by binarymist** [comment]
Ah, sorry... [app-module-path](https://www.npmjs.com/package/app-module-path)

**Comment by benmosher** [comment]
> I'm using app-module-path for my package deps.

Looking briefly at the docs for `app-module-path`, your best bet is to use absolute paths (generally by using some sort of `path.join(__dirname, './config')` incarnation).

FWIW I thing you want `paths: [__dirname]` if I understand your layout. Module resolution will be attempted starting at the path as a root, so to resolve `config/config` you'd be resolving `./config/config/config.js` given your pasted config.

Closing because it is tough to manage config questions but other ideas/responses from the community are still welcome.

**Comment by Venryx** [comment]
@benmosher `path.resolve("Source")` worked for me. (for others who stumble across this thread)
